tests/fixtures/sample_project/pyproject.toml

```diff
@@ -50,6 +50,10 @@ dataclasses = {version = "^0.7", python = ">=3.6.1,<3.7"}
 [tool.poetry.extras]
 db = [ "orator" ]
 
+# Allows testing https://github.com/python-poetry/poetry-core/pull/492
+[tool.black]
```

Comment:
Can you elaborate on the purpose of this bit?

Reply:
Oh yeah, since the issue we had in 1.3.0 is rather fresh in my head, I thought that the link to the revert would give enough context, but not at all.

Basically, the issue in the original PR is that `tomlkit` can either return a `Table` for TOML tables or `OutOfOrderTableProxy` if the table is discontinuous and separated by unrelated tables.
As we explicitly asserted that `tool.poetry` was a `Table`, we ended up raising errors for people having discontinuous sections for `tool.poetry` (who would end up with `OutOfOrderTableProxy` instances).

So this change is mostly a non-regression test to ensure that we don't reintroduce the issue. It might be worth having a dedicated test for it rather than updating an existing fixture used in multiple tests, although this edge case is rather specific to `tomlkit`, which sooner or later will go away, so not sure it's worth doing so.

Regardless, the comment is clearly not explicit enough, so it would need to be updated.